tests/index.test.js

```diff
@@ -0,0 +1,55 @@
+import { map, filter, find, reduce, every } from '../index.js';
+
+describe('map', () => {
+  it('takes an array and a callback function and creates a new array with the callback function implemented', () => {
+    const numbers = [1, 2, 3, 4, 5];
+    const modification = (item) => item * 2;
+    const actual = map(numbers, modification);
+
+    expect(actual).toEqual([2, 4, 6, 8, 10]);
+    expect(numbers).toEqual([1, 2, 3, 4, 5]);
+  });
+});
+
+describe('filter', () => {
+  it('takes an array and callback of signature and creates a new array with all items whose callback returned true', () => {
+    const numbers = [1, 2, 3, 4, 5];
+    const modification = (item) => item > 3;
+    const expected = [4, 5];
+    const actual = filter(numbers, modification);
+
+    expect(expected).toEqual(actual);
+  });
+});
+
+describe('find', () => {
+  it('takes an array and a callback fucntion and returns the index of the found item', () => {
+    const names = ['Angel', 'Missael', 'Maria'];
+    const modification = item => item[0].toLowerCase() === 'm';
```

Comment:
This is a *really* great test! I love that it's not just a simple identity check (like, 'find the thing that === Maria'), and that you have more than one item in the array that could ping as true so you're making sure that it stops at the first one/only returns the first one.